airflow/utils/dag_processing.py

```diff
@@ -696,11 +706,20 @@ def _run_parsing_loop(self):
             max_runs_reached = self.max_runs_reached()
 
             dag_parsing_stat = DagParsingStat(
-                self._file_paths,
+                len(self._file_paths),
                 max_runs_reached,
                 all_files_processed,
             )
-            self._signal_conn.send(dag_parsing_stat)
+            try:
+                self._signal_conn.send(dag_parsing_stat)
```

Comment:
I'm not sure: does applying a timeout context here makes sense?
The reason is it may try to send and get stuck but not fail explicitly.

May be dumb question.

Reply:
Since these are local sockets, with the change to Non-blocking mode, it will either succeed, or fail entirely, "instantly", so we don't need a timeout, no.

Reply:
It wouldn't be a bad idea to add a timeout, but that's extremely rarely an issue for local sockets (things need to get pretty wrong for that to happen at all). This can always be revisited if someone ever complains.

Reply:
On Linux these are AF_UNIX sockets too, so even harder to get it to not get instant fail/success in non-blocking mode I think